### PR TITLE
Do not auto-set extra config that user wishes to override

### LINKED ISSUE #8130: kubelet resolv-conf is not overriding
**Current status:**
OS: Ubuntu 18.0.4
Minikube version: 1.10.x
Driver: none
On former versions, the minikube is auto setting extra-config to "kubelet.resolv-conf=/run/systemd/resolve/resolv.conf" only when kubelet.resolv.conf is not set.
But once it is set, then auto setting is ignored, and the specified conf was overriding.

On version 1.10.x, the minikube is auto setting extra-config to "kubelet.resolv-conf=/run/systemd/resolve/resolv.conf" always.
i.e. the specified resolv conf is appending to the systemd resolv conf.

v1.9.0
$minikube start
```
...
Preparing Kubernetes v1.18.0 on Docker 19.03.8 ...
 - kubelet.resolv-conf= /run/systemd/resolve/resolv.conf 
...
```
$minikube start --extra-config=kubelet.resolv-conf=/home/zuul/.minikube/k8s_resolv.conf

```
...
Preparing Kubernetes v1.18.0 on Docker 19.03.8 ...
 - kubelet.resolv-conf=/home/zuul/.minikube/k8s_resolv.conf
...
```

v1.10.x
$minikube start
```
...
Preparing Kubernetes v1.18.0 on Docker 19.03.8 ...
 - kubelet.resolv-conf= /run/systemd/resolve/resolv.conf 
...
```
$minikube start --extra-config=kubelet.resolv-conf=/home/zuul/.minikube/k8s_resolv.conf

```
...
Preparing Kubernetes v1.18.0 on Docker 19.03.8 ...
 - kubelet.resolv-conf=/home/zuul/.minikube/k8s_resolv.conf
 - kubelet.resolv-conf= /run/systemd/resolve/resolv.conf 
...
```

**Expectation**
kubelet.resolv-conf would be overriding.
So once that is specified, then the auto setting would be ignored.

### PR DESCRIPTION
**Pull request by prasadkatti**
Fixes #8130 

Based on @priyawadhwa's suggestion in https://github.com/kubernetes/minikube/issues/8130#issuecomment-631048823

### CODE REVIEW

**Comment by prasadkatti** [comment]
Hi @priyawadhwa. Would you be willing to review this? I have made the changes you suggested to fix the issue.

File: pkg/minikube/config/extra_options.go [hunk 1]
```diff
@@ -38,6 +40,29 @@ type ExtraOptionSlice []ExtraOption
 // ComponentExtraOptionMap maps components to their extra opts, which is a map of keys to values
 type ComponentExtraOptionMap map[string]map[string]string
 
+// Exists returns true if component.key (parsed from value) is already in ExtraOptionSlice
+func (es *ExtraOptionSlice) Exists(value string) bool {
```
Comment (priyawadhwa) [inline pkg/minikube/config/extra_options.go:44]: This looks good! Would you mind writing a unit test for this in `extra_options_test.go`?

Comment (prasadkatti) [reply pkg/minikube/config/extra_options.go:44]: Added unit tests...

**Pull request merged**